src/Processors/Formats/Impl/NpyRowInputFormat.cpp

```diff
@@ -34,6 +30,43 @@ namespace ErrorCodes
 namespace
 {
 
+float convertFloat16ToFloat32(uint16_t float16_value)
+{
+    uint16_t sign = (float16_value >> 15) & 0x1;
+    uint16_t exponent = (float16_value >> 10) & 0x1F;
+    uint16_t fraction = float16_value & 0x3FF;
+
+    if (exponent == 0 && fraction == 0)
+        return float(sign << 31);
```

Comment:
When sign=1 it will cast `2 ** 31` to float and we will get `-2.14748e+09` instead of `-0.0`. We can use reinterpret_cast:
```suggestion
    {
         uint32_t float32_value = sign << 31;
         return *reinterpret_cast<float *>(&float32_value);
    }
```